### PR TITLE
A Complete LocationView v0.0.1

### PR DESCRIPTION
**Pull request by MohamedSaiko**
Requesting Code Review and any Recommended Updates

AuthenticatedUser
https://github.com/MohamedSaiko/iStore/assets/60336337/f342e09a-9e4f-4190-a1e3-2ba3bd8a2085

Guest
https://github.com/MohamedSaiko/iStore/assets/60336337/ed5f4964-eec5-48a4-964e-1492030735c5

### CODE REVIEW

File: iStore/Home/HomeView.swift [hunk 1]
```diff
@@ -10,13 +10,18 @@ import SwiftUI
...
     
     var body: some View {
-        NavigationView {
+        NavigationStack {
             VStack() {
-                LocationButton(address: authenticationViewModel.user.address.address, postalCode: authenticationViewModel.user.address.postalCode, city: authenticationViewModel.user.address.city, state: authenticationViewModel.user.address.state)
+                if authenticationViewModel.isGuest || locationDataManager.isCurrentLocation {
+                    UserAddressButton(address: locationDataManager.address, postalCode: locationDataManager.postalCode, city: locationDataManager.city, state: locationDataManager.state, latitude: locationDataManager.latitude, longitude: locationDataManager.longitude)
+                } else {
+                    UserAddressButton(address: authenticationViewModel.user.address.address, postalCode: authenticationViewModel.user.address.postalCode, city: authenticationViewModel.user.address.city, state: authenticationViewModel.user.address.state, latitude: authenticationViewModel.user.address.coordinates.lat, longitude: authenticationViewModel.user.address.coordinates.lng)
```
Comment (akabel92) [inline iStore/Home/HomeView.swift:23]: Lets talk about this in a call.

File: iStore/Location/LocationDataManager.swift [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        case .restricted:
+            break
+            
+        case .denied:
+            break
+            
+        case .notDetermined:
+            locationManager.requestWhenInUseAuthorization()
+            
+        default:
```
Comment (akabel92) [inline iStore/Location/LocationDataManager.swift:42]: you can make it 
`case .restricted, .denied:
     break
  `

Comment (akabel92) [reply iStore/Location/LocationDataManager.swift:42]: Also you can remove both case and only keep `default` case with break

File: iStore/Location/LocationDataManager.swift [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+            break
+        }
+        
+        locationManager.startUpdatingLocation()
+    }
+    
+    func getUserAddress(completion: @escaping (CLPlacemark) -> Void) {
+        if let Location = self.locationManager.location {
```
Comment (akabel92) [inline iStore/Location/LocationDataManager.swift]: Why `if let` here not `guard`?

File: iStore/Location/LocationDataManager.swift [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+                guard let address = placemarks?.first, error == nil else {
+                    return
+                }
+                completion(address)
+            }
+        }
+    }
+}
+
+extension LocationDataManager: CLLocationManagerDelegate {
```
Comment (akabel92) [inline iStore/Location/LocationDataManager.swift:67]: Please add `// Mark` line above for better reading

File: iStore/Home/HomeView.swift (hunk 1 above)
Comment (MohamedSaiko) [reply iStore/Home/HomeView.swift:23]: Ok, call me when you are free.

File: iStore/Location/LocationDataManager.swift (hunk 3 above)
Comment (MohamedSaiko) [reply iStore/Location/LocationDataManager.swift]: I think both are applicable.
but guard let is better as It returns earlier if there is an error.

**Review by akabel92 — APPROVED** [review]

**Pull request merged**